bin/sync/article_sync_checker.rb

```
@@ -0,0 +1,74 @@
+    data = @yaml_parser.load_file(SYNC_STATUS_FILE)
+    data || {}
+  end
+
+  def save_sync_status
+    @file_manager.open(SYNC_STATUS_FILE, 'w') do |file|
+      file.write(@sync_status.to_yaml)
+    end
+  end
+
+  def fetch_articles
+    uri = URI(API_URL + "&per_page=#{999..9999}") # hack to avoid cached API response
```

rubocop

[warning] Literal interpolation detected.

(Lint/LiteralInInterpolation)